networkx/algorithms/approximation/tests/test_steinertree.py

```diff
@@ -0,0 +1,54 @@
+import networkx as nx
+from networkx.algorithms.approximation.steinertree import metric_closure
+from networkx.algorithms.approximation.steinertree import steiner_tree
+from networkx.testing.utils import assert_edges_equal
+
+
+class TestSteinerTree:
+    def setUp(self):
+        G = nx.Graph()
+        G.add_edge(1, 2, weight=1)
+        G.add_edge(2, 3, weight=1)
+        G.add_edge(3, 4, weight=1)
+        G.add_edge(4, 5, weight=1)
+        G.add_edge(5, 6, weight=1)
+        G.add_edge(2, 7, weight=0.1)
+        G.add_edge(7, 5, weight=0.1)
+        self.G = G
+        self.term_nodes = [1, 2, 3, 4, 5]
+
+    def test_metric_closure(self):
+        M = metric_closure(self.G)
+        mc = [(1, 2, {'distance': 1, 'path': [1, 2]}),
+              (1, 3, {'distance': 2, 'path': [1, 2, 3]}),
+              (1, 4, {'distance': 2.2, 'path': [1, 2, 7, 5, 4]}),
+              (1, 5, {'distance': 1.2000000000000002, 'path': [1, 2, 7, 5]}),
```

Comment:
Those values are obtained from single_source_dijkstra. I think it is because of floating point arithmetic. https://docs.python.org/2/tutorial/floatingpoint.html .Not sure though.